flask_smorest/globals.py

```diff
@@ -0,0 +1,20 @@
+from flask import current_app, has_request_context, request
+from werkzeug.local import LocalProxy
+
+
+def _find_current_api():
+    if has_request_context() and "flask-smorest" in current_app.extensions:
+        apis = current_app.extensions["flask-smorest"]["apis"].values()
+        blueprints = set(request.blueprints)
+        for api in (x["ext_obj"] for x in apis):
+            if api._registered_blueprint_names & blueprints:
+                return api
+    return None
```

Comment:
> Would it make sense to set a {blp_name: api_obj} mapping in app.extensions["flask-smorest"] when registering blueprints so that the lookup here can be replaced with a simple key access? Or does it sound too hackish?

I think it is better than a set and for-eaching through all apis. I've updated PR to use `{blp_name: api_obj}` instead. Thanks.

> Could blueprint name clash occur?

No. It raises ValueError.

https://github.com/pallets/flask/blob/main/src/flask/blueprints.py#L383-L391
```python
        if name in app.blueprints:
            ...
            raise ValueError(f"The name '{self_name}' is already registered...")
```

> I expected this part to be simpler but it was just an intuition. I haven't been digging into this myself, so maybe that's how it is.

It is annoying because a blueprint can be registered with multiple APIs. Sometimes even indirectly. For example: you can register a blueprint with one API and then nest it into other blueprint. Which in turn would be registered with some another API.

So I can not mess with a passed blueprint. Which means that I would need to add some kind of `BlueprintProxy` to intercept requests, set `current_api`, and pass them to an original blueprint. Would work I guess. But it would mean that each API request would have an additional function call. And I do not think that `current_api` would be needed 99% of time. So I would rather make getting `current_api` slightly slower than to add an additional frame stack to each API request.